### PR TITLE
function to add new skill

### PR DESCRIPTION
**Pull request by john-thuo1**
#16

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -75,6 +75,27 @@ def skill():
...
+
+def add_skill():
+    
+    '''
+     Add a skill using POST method
+     
+    '''
+    req = request.get_json()
+
+    required_fields = {"name":"string", "proficiency":"string", "logo":"string"}
```
Comment (fissoreg) [inline app.py:91]: Here you are manually repeating the fields from the Skill dataclass, which might be error prone. In general, it is a good idea to have only one source of information and always refer to it. I think you can achieve this with the [`.fields`](https://docs.python.org/3/library/dataclasses.html#dataclasses.fields) method of dataclasses. Can you look into that and let me know if this is feasible? :)

File: app.py [hunk 2]
```diff
@@ -75,6 +75,27 @@ def skill():
...
+
+    missing_fields = [field for field in required_fields.keys() if field not in req]
+    
+    if missing_fields:
+        return jsonify({"error": "Missing required field(s): " + ", ".join(missing_fields)}), 400
```
Comment (fissoreg) [inline app.py:96]: Great stuff, nice to see you are checking for errors! :)
The line results long and difficult to parse; maybe you can try to split it on multiple lines?

**Review by fissoreg — CHANGES_REQUESTED** [review]
This is great, thank you for the contribution! :)

I left a couple of comments for you to review. Also, linting and tests are not passing. Can you look into that?

File: app.py (hunk 1 above)
Comment (john-thuo1) [reply app.py:91]: I am thinking of using a list comprehension to get the required fields from the skill dataclass. Something akin to 
  > required_fields = [field. name for field in fields(Skill)]. I have tried it, and it works. Will this do?

Comment (john-thuo1) [reply app.py]: On the Errors, I have managed to fix the pylint error and some of the pytest test cases, notably _test_skill() & test_client()._ The remaining 2 test cases are associated with education and experience whose functionalities are yet to be implemented.

**Comment by fissoreg** [comment]
LGTM, congrats!

**Comment by fissoreg** [comment]
Close #16

**Pull request merged**